### PR TITLE
Moves saving of bd files from php to flask

### PR DESCRIPTION
**Pull request by cabarnes**
This pull request addresses issue #22

### CODE REVIEW

**Comment by aninternetof** [comment]
With every save, I get this from Flask:

```
127.0.0.1 - - [20/Sep/2016 11:43:06] "POST /api/v1/blockdiagrams HTTP/1.1" 302
127.0.0.1 - - [20/Sep/2016 11:43:06] "GET /api/v1/blockdiagrams HTTP/1.1" 200 -
```

Were we expecting to get a GET following every post? Or is this something in my old JS that I've forgotten about?

**Comment by aninternetof** [comment]
Hmm.. I did a test by making a new bd named "save test". I added some things to the design, and it seemed to be saving with no errors. I then refreshed the page and tried to reload the design and got this from Flask:

```
File "C:\Users\blhurlburt\Personal_Projects\rover-code\rover-code\www\app.py", line 35, in getblockdiagram
with open(join('saved-bds',bd[0]), 'r') as content_file:
IndexError: list index out of range
```

I tested again with a file I had saved using the development branch, it it opened ok.

**Comment by aninternetof** [comment]
Oh, I think I found the issue! The design had a space in the name. This should be replaced with an underscore before saving. (Or something else if you have other ideas)

File: www/app.py [hunk 1]
```diff
@@ -15,6 +15,19 @@ def getblockdiagrams():
             names.append(xml.etree.ElementTree.parse(join('saved-bds', f)).getroot().find('designName').text)
     return jsonify(result=names)
 
+@app.route('/api/v1/blockdiagrams', methods=['POST'])
+def saveblockdiagram():
+    designName = request.form['designName']
+    bdString = request.form['bdString']
+    root = xml.etree.ElementTree.Element("root")
+
+    xml.etree.ElementTree.SubElement(root, 'designName').text = designName
```
Comment (aninternetof) [inline www/app.py]: Use

`xml.etree.ElementTree.SubElement(root, 'designName').text = designName.replace(' ', '_').replace('.', '_')`

File: www/app.py [hunk 2]
```diff
@@ -15,6 +15,19 @@ def getblockdiagrams():
...
+    xml.etree.ElementTree.SubElement(root, 'bd').text = bdString
+
+    tree = xml.etree.ElementTree.ElementTree(root)
+    tree.write('saved-bds/' + designName + '.xml')
+    return redirect(request.url)
```
Comment (wavetylor) [inline www/app.py]: what's going on with returning a redirect?

File: www/app.py (hunk 1 above)
Comment (cabarnes) [inline www/app.py]: Ah, I hadn't tried with a space. I'll make the change.

File: www/app.py (hunk 2 above)
Comment (cabarnes) [inline www/app.py]: I wasn't sure what to return here from the POST. We don't want to change the page after a save. Is there a way to just send a 200 OK without anything else?

Comment (wavetylor) [inline www/app.py]: I think this is all you need. Its an empty response body with a 200 OK.
`return ('', 200)`

Comment (aninternetof) [inline www/app.py]: This change seems to have gotten rid of the extraneous GET after every POST, too.

**Pull request merged**